pandas/core/strings.py

```diff
@@ -1020,7 +1024,17 @@ def str_extractall(arr, pat, flags=0):
 
     index = MultiIndex.from_tuples(index_list, names=arr.index.names + ["match"])
 
-    result = arr._constructor_expanddim(match_list, index=index, columns=columns)
+    # workaround #27953
+    # ideally we just pass `dtype=arr.dtype` unconditionally, but this fails
+    # when the list of values is empty.
+    if arr.dtype.name == "string":
+        dtype = arr.dtype
+    else:
+        dtype = None
+
+    result = arr._constructor_expanddim(
```

Comment:
does arr.dtype work here?

Reply:
Not always: https://github.com/pandas-dev/pandas/issues/27953